phpunit/experimental/interactivity-api/directive-processing-test.php

```diff
@@ -73,73 +73,43 @@ public function test_directives_with_double_hyphen_processed_correctly() {
 		'<!-- /wp:paragraph -->' .
 		'<!-- wp:paragraph -->' .
-			'<p>Pattern Content Block 2</p>' .
+			'<p>Welcome to WordPress.</p>' .
 		'<!-- /wp:paragraph -->';
-		register_block_pattern(
-			'core/interactivity-pattern',
-			array(
-				'title'   => 'Interactivity Pattern',
-				'content' => $pattern_content,
-			)
-		);
 
-		$providers = $this->data_only_root_blocks_are_processed();
-		foreach ( $providers as $provider ) {
-			do_blocks( $provider['page_content'] );
-			$this->assertSame( $provider['root_blocks'], count( WP_Directive_Processor::$root_blocks ) );
+		$parsed_block     = parse_blocks( $block_content )[0];
+		$rendered_content = render_block( $parsed_block );
 
-		}
-	}
+		$parsed_block_second = parse_blocks( $block_content )[1];
 
-	/**
-	 * Data provider .
-	 *
-	 * @return array
-	 **/
-	public function data_only_root_blocks_are_processed() {
+		// Test that root block is intially emtpy.
+		$this->assertEmpty( WP_Directive_Processor::$root_block );
 
-		return array(
-			array(
-				'root_blocks'  => 2,
-				'page_content' =>
-				'<!-- wp:quote -->' .
-					'<blockquote class="wp-block-quote">
-						<!-- wp:paragraph -->' .
-						'<p>The XYZ Doohickey Company was founded in 1971, and has been providing' .
-						'quality doohickeys to the public ever since. Located in Gotham City, XYZ employs' .
-						'over 2,000 people and does all kinds of awesome things for the Gotham community.</p>' .
-						'<!-- /wp:paragraph -->
-					</blockquote>' .
-				'<!-- /wp:quote -->' .
-				'<!-- wp:quote -->' .
-					'<blockquote class="wp-block-quote">
-						<!-- wp:paragraph -->' .
-						'<p>The XYZ Doohickey Company was founded in 1971, and has been providing' .
-						'quality doohickeys to the public ever since. Located in Gotham City, XYZ employs' .
-						'over 2,000 people and does all kinds of awesome things for the Gotham community.</p>' .
-						'<!-- /wp:paragraph -->
-					</blockquote>' .
-				'<!-- /wp:quote -->',
-			),
-			array(
-				'root_blocks'  => 2,
-				'page_content' =>
-				'<!-- wp:paragraph -->' .
-					'<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>' .
-				'<!-- /wp:paragraph -->' .
-				'<!-- wp:pattern {"slug":"core/interactivity-pattern"} /-->',
-			),
-		);
+		// Test that root block is not added if there is a parent block.
+		gutenberg_interactivity_process_directives( $parsed_block, $parsed_block, $parsed_block );
+		$this->assertEmpty( WP_Directive_Processor::$root_block );
+
+		// Test that root block is added if there is no parent block.
+		gutenberg_interactivity_process_directives( $parsed_block, $parsed_block, null );
+		$current_root_block = WP_Directive_Processor::$root_block;
+		$this->assertNotEmpty( $current_root_block );
+
+		// Test that a root block is not added if there is already a root block defined.
+		gutenberg_interactivity_process_directives( $parsed_block_second, $parsed_block_second, $parsed_block_second );
```

Comment:
You are not testing root blocks if you are passing a parent.

```suggestion
		// Test that a root block is not added if there is already a root block defined.
		gutenberg_interactivity_process_directives( $parsed_block_second, $parsed_block_second, null );
```